pkg/cri/cri.go

```diff
@@ -17,75 +17,59 @@
 package cri
 
 import (
-	"flag"
 	"fmt"
-	"path/filepath"
 
 	"github.com/containerd/log"
 	"github.com/containerd/plugin"
 	"github.com/containerd/plugin/registry"
-	imagespec "github.com/opencontainers/image-spec/specs-go/v1"
-	"k8s.io/klog/v2"
 
 	containerd "github.com/containerd/containerd/v2/client"
-	criconfig "github.com/containerd/containerd/v2/pkg/cri/config"
 	"github.com/containerd/containerd/v2/pkg/cri/constants"
 	"github.com/containerd/containerd/v2/pkg/cri/nri"
 	"github.com/containerd/containerd/v2/pkg/cri/server"
+	"github.com/containerd/containerd/v2/pkg/cri/server/base"
+	"github.com/containerd/containerd/v2/pkg/cri/server/images"
 	nriservice "github.com/containerd/containerd/v2/pkg/nri"
 	"github.com/containerd/containerd/v2/platforms"
 	"github.com/containerd/containerd/v2/plugins"
-	"github.com/containerd/containerd/v2/services/warning"
 )
 
 // Register CRI service plugin
 func init() {
-	config := criconfig.DefaultConfig()
+
 	registry.Register(&plugin.Registration{
-		Type:   plugins.GRPCPlugin,
-		ID:     "cri",
-		Config: &config,
+		Type: plugins.GRPCPlugin,
+		ID:   "cri-grpc-service",
```

Comment:
could you just keep it as `cri`? It's unlikely to do live-migration because the running containers are using the volume in this plugin folder.

Reply:
We probably need to add some release upgrade integration cases for this before we accept this. 

REF: https://github.com/containerd/containerd/blob/6ed8f01aa1472c9e6179e860f30cb9b6d3977b3d/integration/release_upgrade_linux_test.go#L52

Reply:
> could you just keep it as `cri`? It's unlikely to do live-migration because the running containers are using the volume in this plugin folder.

That is actually why the intergration test failed before. So that I add a commit [9a18e8e](https://github.com/containerd/containerd/pull/9391/commits/9a18e8e9396b5f950c379374ac77e8bb2f780741) to keep the plugin folder path "io.containerd.grpc.v1.cri" as before.

I think it is no use to change the name here because we want to add a plugin of type io.containerd.internal.v1 to carry the cri configs and other plugins will be depend on this plugin to get the cri configs.

Reply:
> We probably need to add some release upgrade integration cases for this before we accept this.

added a test case to manipulate the containers in the pod before upgrade.  [0597be4849](https://github.com/containerd/containerd/pull/9391/commits/0597be48493676107da2d00f89c692d7e622f871)

Reply:
The added integration test case reveals an existing compatibility issues, and I fixed in [e18e86905b3071](https://github.com/containerd/containerd/pull/9391/commits/e18e86905b307122cab1acbd14e02b27a4e731b9), all these are included in this PR, could you please take a look at the new commits ?

or, as Max has approved, shall I put the two new commits in a new PR?  @mxpv @fuweid